core/src/main/java/org/apache/carbondata/core/carbon/datastore/block/Distributable.java

```diff
@@ -16,10 +16,12 @@
  */
 package org.apache.carbondata.core.carbon.datastore.block;
 
+import java.io.IOException;
+
 /**
- * Abstract class which is maintains the locations of node.
+ * interface to get the locations of node. Used for making task distribution based on locality
  */
-public abstract class Distributable implements Comparable<Distributable> {
+public interface Distributable extends Comparable<Distributable> {
 
-  public abstract String[] getLocations();
+  String[] getLocations() throws IOException;
```

Comment:
Any reason to throw IOException form this method, I think this is not required ??